analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/NugetMetadataFactory.cs

```diff
@@ -210,7 +210,7 @@ private static IEnumerable<string> GetSortedPackageFolders(string packageId)
             //      e.g. Microsoft.AspNetCore.Core and Microsoft.AspNetCore.Core.Diagnostics
             // Most packages have a three-part version, but some have four. We don't check
             // the actual number of parts, as long as there is at least one.
-            var matcher = new Regex($"{packageId}(.\\d+)+$");
+            var matcher = new Regex($@"{Regex.Escape(packageId)}(\.\d+)+$", RegexOptions.IgnoreCase);
```

Comment:
Oh boy, we need those regex rules. This one was doing `packageId followed by anything that contains at least one digit` (it was fine with `NCover3Package` suffix as well) instead of `packageId followed by sequence of a dot followed by one or more digits`.

Reply:
and we need to start analyzing test code, because even if we had regex rules, they wouldn't run on this project